lib/onelogin/ruby-saml/logoutresponse.rb

```diff
@@ -3,50 +3,61 @@
 
 require "time"
 
+# Only supports SAML 2.0
 module OneLogin
   module RubySaml
+
+    # SAML2 Logout Response (SLO IdP initiated, Parser)
+    #
     class Logoutresponse < SamlMessage
-      # For API compability, this is mutable.
+
+      # OneLogin::RubySaml::Settings Toolkit settings
       attr_accessor :settings
 
-      attr_reader :document
-      attr_reader :response
+      # Array with the causes
+      attr_accessor :errors
+
       attr_reader :options
+      attr_reader :response
+      attr_reader :document
+      attr_reader :get_params
 
+      # Constructs the Logout Response. A Logout Response Object that is an extension of the SamlMessage class.
+      # @param response   [String] A UUEncoded logout response from the IdP.
+      # @param settings   [OneLogin::RubySaml::Settings|nil] Toolkit settings
+      # @raise [ArgumentError]
       #
-      # In order to validate that the response matches a given request, append
-      # the option:
-      #   :matches_request_id => REQUEST_ID
-      #
-      # It will validate that the logout response matches the ID of the request.
-      # You can also do this yourself through the in_response_to accessor.
-      #
-      def initialize(response, settings = nil, options = {})
+      def initialize(response, settings = nil)
+        @errors = []
         raise ArgumentError.new("Logoutresponse cannot be nil") if response.nil?
-        self.settings = settings
+        @settings = settings
 
         @options = options
         @response = decode_raw_saml(response)
         @document = XMLSecurity::SignedDocument.new(@response)
-      end
 
-      def validate!
-        validate(false)
       end
 
-      def validate(soft = true)
-        return false unless valid_saml?(document, soft) && valid_state?(soft)
-
-        valid_in_response_to?(soft) && valid_issuer?(soft) && success?(soft)
+      # An aux function to validate the Logout Response with the default values (soft = true)
+      # @return [Boolean] TRUE if the Logout Response is valid
+      #
+      def is_valid?
+        validate
       end
 
-      def success?(soft = true)
-        unless status_code == "urn:oasis:names:tc:SAML:2.0:status:Success"
-          return soft ? false : validation_error("Bad status code. Expected <urn:oasis:names:tc:SAML:2.0:status:Success>, but was: <#@status_code> ")
-        end
-        true
-      end
+      # Another aux function to validate the Logout Response (soft = false)
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @param request_id [String|nil] request_id The ID of the Logout Request sent by this SP to the IdP (if was sent any)
+      # @param get_params [Hash] In order to validate the In value we need to provide the GET parameters
 
+      # @return [Boolean] TRUE if the SAML Response is valid
+      #
+      def validate!(soft=false, request_id = nil, get_params = nil)
```

Comment:
The Response method got it, so I added this to the logoutresponse too (compatibility).
Do you think we may remove it?